lib/Twig/Error.php

```diff
@@ -51,13 +54,23 @@ class Twig_Error extends Exception
      *
      * By default, automatic guessing is enabled.
      *
-     * @param string    $message  The error message
-     * @param int       $lineno   The template line where the error occurred
-     * @param string    $name     The template logical name where the error occurred
-     * @param Exception $previous The previous exception
+     * @param string      $message  The error message
+     * @param int         $lineno   The template line where the error occurred
+     * @param Twig_Source $source   The source context where the error occurred
```

Comment:
``Twig_Source|null`` (or ``Twig_Source|string|null`` if we don't deprecate the case of passing a string)